### PR TITLE
CM_BE_26: Provider by ID

### PR DESCRIPTION
**Pull request by Veena-tw**
Removes hip name and hiu name from consent-request, consent-artefact, link related APIs

Introduces a new API to fetch provider information by ID

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5388016&bid=17665284)

File: src/main/java/in/projecteka/consentmanager/link/link/Link.java [hunk 1]
```diff
@@ -129,31 +123,7 @@
...
-                        .patientRepresentations(links.getPatientRepresentations())
-                        .build());
-    }
-
-    private Mono<String> getProviderName(String providerId) {
-        return centralRegistry.providerWith(providerId)
-                .map(Provider::getName)
-                .switchIfEmpty(Mono.empty());
+        var patientLinks = linkRepository.getLinkedCareContextsForAllHip(patientId).block();
+        return Mono.just(PatientLinksResponse.builder().patient(patientLinks).build());
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/link/link/Link.java:127]: you should never block it.

Comment (Veena-tw) [reply src/main/java/in/projecteka/consentmanager/link/link/Link.java]: removed.

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5388016&bid=17665284)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5388016&bid=17665284)

File: src/main/java/in/projecteka/consentmanager/link/link/Link.java [hunk 2]
```diff
@@ -107,31 +100,7 @@
...
-                        .patientRepresentations(links.getPatientRepresentations())
-                        .build());
-    }
-
-    private Mono<String> getProviderName(String providerId) {
-        return centralRegistry.providerWith(providerId)
-                .map(Provider::getName)
-                .switchIfEmpty(Mono.empty());
+        return linkRepository.getLinkedCareContextsForAllHip(patientId).flatMap(patientLinks ->
+                Mono.just(PatientLinksResponse.builder().patient(patientLinks).build()));
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/link/link/Link.java:104]:
```suggestion
        return linkRepository.getLinkedCareContextsForAllHip(patientId)
        .map(patientLinks -> PatientLinksResponse.builder().patient(patientLinks).build());
```

Comment (Veena-tw) [reply src/main/java/in/projecteka/consentmanager/link/link/Link.java]: done.

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5388016&bid=17665284)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5388016&bid=17665284)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5388016&bid=17665284)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5388016&bid=17665284)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5388016&bid=17665284)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5388016&bid=17665284)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5388016&bid=17665284)

**Pull request merged**